app/connector/FrontendAuthConnector.scala

```diff
@@ -50,19 +52,19 @@ object FrontendAuthConnector extends FrontendAuthConnector with ServicesConfig {
 
 trait FrontendAuthConnector extends AuthConnector {
 
-  private case class Authority(twoFactorAuthOtpId: Option[String], enrolmentsUri: Option[String], userDetailsLink: String, idsUri: String)
+  private case class Authority(twoFactorAuthOtpId: Option[String], enrolmentsUri: Option[String], userDetailsLink: Option[String], idsUri: Option[String])
 
   private object Authority {
     implicit val reads: Reads[Authority] =
       ((__ \ "twoFactorAuthOtpId").readNullable[String] and
         (__ \ "enrolments").readNullable[String] and
-        (__ \ "userDetailsLink").read[String] and
-        (__ \ "ids").read[String]).apply(Authority.apply _)
+        (__ \ "userDetailsLink").readNullable[String] and
+        (__ \ "ids").readNullable[String]).apply(Authority.apply _)
   }
 
   def currentCoAFEAuthority()(implicit hc: HeaderCarrier) = {
 
-    def getIds(idsUri: String) = http.GET[InternalUserIdentifier](s"$serviceUrl$idsUri")
+    def getIds(idsUri: Option[String]) = idsUri.map(uri => http.GET[InternalUserIdentifier](s"$serviceUrl$uri").map(Option(_))).getOrElse(Future.successful(None))
```

Comment:
As discussed the getIds might get just the idsUri and should stay dummy in that way it should not care with checking if it is None or Some. 
Such logic should be outside (ideally in RuleContext) and the connector call should only be executed if there is an idsUri.

See ruleContext.enrolment implementation for an example.